pulser/simulation/simresults.py

```diff
@@ -59,6 +64,192 @@ def __init__(self, run_output: Sequence[qutip.Qobj], dim: int, size: int,
                     "`meas_basis` must be 'ground-rydberg' or 'digital'."
                     )
         self._meas_basis = meas_basis
+        self.sim_times = sim_times
+
+
+class NoisyResults(SimulationResults):
+    """Results of a noisy simulation run of a pulse sequence.
+    Contrary to a CleanResults object, this object contains a list of Counter
+    describing the state distribution at the time it was created by using
+    Simulation.run() with a noisy simulation.
+    Contains methods for studying the populations and extracting useful
+    information from them.
+    """
+
+    def __init__(self, run_output: Sequence[Counter],
+                 size: int, basis_name: str, meas_basis: str,
+                 sim_times: ArrayLike, N_measures: int, dim: int = 2) -> None:
+        """Initializes a new NoisyResults instance.
+        Warning :
+            Can't have single-atom Hilbert spaces with dimension bigger
+            than 2 for NoisyResults objects.
+            This is not the case for a CleanResults object, containing states
+            in Hilbert space, but NoisyResults contains a probability
+            distribution of bitstrings, not atomic states
+        Args:
+            run_output (Counter List): Each Counter contains the
+                probability distribution of a multi-qubits state,
+                represented as a bitstring. There is one Counter for each time
+                the simulation was asked to return a result.
+            dim (int): The dimension of the local space of each atom (2 or 3).
+            size (int): The number of atoms in the register.
+            basis_name (str): The basis indicating the addressed atoms after
+                the pulse sequence ('ground-rydberg', 'digital' or 'all').
+            sim_times (list): times at which Simulation object returned the
+                results.
+        Keyword Args:
+            meas_basis (None or str): The basis in which a sampling measurement
+                is desired.
+            N_measures (int): number of measurements needed to compute this
+                result when doing the simulation.
+        """
+        super().__init__(run_output, dim, size, basis_name, meas_basis,
+                         sim_times)
+        self.N_measures = N_measures
+
+    @property
+    def states(self) -> Sequence[Counter]:
+        """Probability distribution of the bitstrings"""
+        return self._states
```

Comment:
It's a bit misleading to call these `states`, don't you think? I'm not saying to change it right away, but perhaps we can think of a better solution